src/include/plan_node/aggregate_plan_node.h

```diff
@@ -0,0 +1,203 @@
+     * Predicate for having clause if it exists
+     */
+    std::unique_ptr<const parser::AbstractExpression> having_clause_predicate_;
+    /**
+     * List of aggregate terms for aggregation
+     */
+    std::vector<AggregateTerm> aggregate_terms_;
+    /**
+     * Strategy to use for aggregation
+     */
+    AggregateStrategy aggregate_strategy_;
+  };
+
+  /**
+   * @param children child plan nodes
+   * @param output_schema Schema representing the structure of the output of this plan node
+   * @param estimated_cardinality estimated cardinality of output of node
+   * @param having_clause_predicate unique pointer to possible having clause predicate
+   * @param aggregate_terms vector of aggregate terms for the aggregation
+   * @param aggregate_strategy aggregation strategy to be used
+   */
+  AggregatePlanNode(std::vector<std::unique_ptr<AbstractPlanNode>> &&children,
+                    std::shared_ptr<OutputSchema> output_schema, uint32_t estimated_cardinality,
+                    std::unique_ptr<const parser::AbstractExpression> &&having_clause_predicate,
+                    std::vector<AggregateTerm> aggregate_terms, AggregateStrategy aggregate_strategy)
+      : AbstractPlanNode(std::move(children), std::move(output_schema), estimated_cardinality),
+        having_clause_predicate_(std::move(having_clause_predicate)),
+        aggregate_terms_(std::move(aggregate_terms)),
+        aggregate_strategy_(aggregate_strategy) {}
+
+ public:
+  ~AggregatePlanNode() override {
+    for (auto term : aggregate_terms_) {
+      delete term.expression_;
+    }
+  }
+
+  //===--------------------------------------------------------------------===//
+  // ACCESSORS
+  //===--------------------------------------------------------------------===//
+
+  // TODO(Gus,Wen): Figure out how to represent global aggregates (example: count(*))
+  /**
+   * A global aggregate does not aggregate over specific columns, example: count(*)
+   * @return true if aggregation is global
+   */
+  bool IsGlobal() const { return false; }
+
+  /**
+   * @return pointer to predicate for having clause
+   */
+  const parser::AbstractExpression *GetHavingClausePredicate() const { return having_clause_predicate_.get(); }
+
+  /**
+   * @return vector of aggregate terms
+   */
+  const std::vector<AggregateTerm> &GetAggregateTerms() const { return aggregate_terms_; }
+
+  /**
+   * @return aggregation strategy
+   */
+  AggregateStrategy GetAggregateStrategy() const { return aggregate_strategy_; }
+
+  /**
+   * @return the type of this plan node
+   */
+  PlanNodeType GetPlanNodeType() const override { return PlanNodeType::AGGREGATE; }
+
+  /**
+   * @return the hashed value of this plan node
+   */
+  common::hash_t Hash() const override;
+
+  bool operator==(const AbstractPlanNode &rhs) const override;
+
+ private:
+  /**
+   * @return true of two vectors of aggregate terms are equal
+   */
+  bool AreEqual(const std::vector<AggregatePlanNode::AggregateTerm> &A,
```

Comment:
I feel like this should be in a static util class?

Reply:
I decided to implement `operator==` for `AggregateTerm` instead. The method was only being used inside the plan node anyway.